dotnet/src/Connectors/Connectors.MistralAI/MistralAIKernelBuilderExtensions.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel.Connectors.MistralAI;
+
+/// <summary>
+/// Provides extension methods for the <see cref="IKernelBuilder"/> class to configure Mistral connectors.
+/// </summary>
+public static class MistralAIKernelBuilderExtensions
+{
+    /// <summary>
+    /// Adds an Mistral chat completion service with the specified configuration.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> instance to augment.</param>
+    /// <param name="model">The name of the Mistral model.</param>
+    /// <param name="endpoint">The endpoint URL for the chat completion service.</param>
+    /// <param name="apiKey">The API key required for accessing the Mistral service.</param>
+    /// <param name="serviceId">A local identifier for the given AI service.</param>
+    /// <returns>The same instance as <paramref name="services"/>.</returns>
+    public static IServiceCollection AddMistralChatCompletion(
+        this IServiceCollection services,
+        string model,
+        Uri? endpoint = null,
+        string? apiKey = null,
+        string? serviceId = null)
+    {
+        Verify.NotNull(services);
+        Verify.NotNull(model);
+
+        return services.AddKeyedSingleton<IChatCompletionService>(serviceId, (serviceProvider, _) =>
+            new MistralAIChatCompletionService(model, endpoint, apiKey, HttpClientProvider.GetHttpClient(serviceProvider)));
+    }
+
+    /// <summary>
+    /// Adds an Mistral chat completion service with the specified configuration.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> instance to augment.</param>
+    /// <param name="model">The name of the Mistral model.</param>
+    /// <param name="endpoint">The endpoint URL for the chat completion service.</param>
+    /// <param name="apiKey">The API key required for accessing the Mistral service.</param>
+    /// <param name="serviceId">A local identifier for the given AI service.</param>
+    /// <param name="httpClient">The HttpClient to use with this service.</param>
+    /// <returns>The same instance as <paramref name="services"/>.</returns>
+    public static IServiceCollection AddMistralChatCompletion(
+        this IServiceCollection services,
```

Comment:
```suggestion
    public static IKernelBuilder AddMistralChatCompletion(
        this IKernelBuilder services,
```

Reply:
I've split the extension methods into two classes

Reply:
But do you need serviceCollection overload with httpClient param? @dmytrostruk said here https://github.com/microsoft/semantic-kernel/pull/5094#discussion_r1507752216 we are giving it up.